### PR TITLE
Electronics

### PR DESCRIPTION
**Pull request by sgasioro**
Rewrites to make the simulation differentiable through to the stage of ADC counts. Parameter fitting notebook updated to be done via comparisons at the ADC count level. Note that there is now a sampling done, leading to some spread in output for a given parameter set. 

Agreement generally good with non-differentiable sim, though still residual difference noticed as of #1. Some small tweaks also included to ensure trivial running on GPUs.

### CODE REVIEW

**Review by ynashed — CHANGES_REQUESTED** [review]
It'll probably be cleaner if we remove all mention of `numba` in this file. Also, the functions that do not contribute to the forward model should be removed (not repeated verbatum).

File: larndsim/fee_ep.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    if x_axis < 0:
+        pix_x = consts.n_pixels_per_tile[0]-pixel_id[0]-1
+        
+    pix_y = pixel_id[1]
+    if y_axis < 0:
+        pix_y = consts.n_pixels_per_tile[1]-pixel_id[1]-1
+    
+    return pix_x, pix_y
+        
+def export_to_hdf5(adc_list, adc_ticks_list, unique_pix, track_ids, filename):
```
Comment (ynashed) [inline larndsim/fee_ep.py:42]: maybe remove

File: larndsim/fee_ep.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+nonrouted_channels=[6,7,8,9,22,23,24,25,38,39,40,54,55,56,57]
+routed_channels=[i for i in range(64) if i not in nonrouted_channels]
+top_row_channels=[3,2,1,63,62,61,60]
+bottom_row_channels=[28,29,30,31,33,34,35]
+inner_edge_channels=[60,52,53,48,45,41,35]
+top_row_chip_ids=[11,12,13,14,15,16,17,18,19,20]
+bottom_row_chip_ids=[101,102,103,104,105,106,107,108,109,110]
+inner_edge_chip_ids=[20,30,40,50,60,70,80,90,100,110]
+
+def rotate_tile(pixel_id, tile_id):
```
Comment (ynashed) [inline larndsim/fee_ep.py:27]: maybe remove

File: larndsim/pixels_from_track_ep.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
 import eagerpy as ep
 import numpy as np
+import torch
```
Comment (ynashed) [inline larndsim/pixels_from_track_ep.py:9]: do we need this?

File: larndsim/pixels_from_track_ep.py [hunk 4]
```diff
@@ -124,7 +125,7 @@ def get_neighboring_pixels(self, active_pixels, radius, max_pixels):
             n_indices = ep.tile(neighbor_indices, [track.shape[0], 1]) + \
                         ep.tile(track, [1, neighbor_indices.shape[0]]).reshape([-1, 2])
             n_indices = n_indices[:, 0] * self.n_pixels[1] + n_indices[:, 1]
-            n_indices_idx =np.unique(n_indices.raw.numpy(), return_index=True)[1]
+            n_indices_idx =np.unique(n_indices.raw.cpu().numpy(), return_index=True)[1]
```
Comment (ynashed) [inline larndsim/pixels_from_track_ep.py:128]: Let's think of an alternative.

File: larndsim/fee_ep.py (hunk 1 above)
Comment (sgasioro) [reply larndsim/fee_ep.py]: Done!

File: larndsim/fee_ep.py (hunk 2 above)
Comment (sgasioro) [reply larndsim/fee_ep.py]: Done!

File: larndsim/pixels_from_track_ep.py (hunk 3 above)
Comment (sgasioro) [reply larndsim/pixels_from_track_ep.py]: Nope, sorry -- switched over the unique call to torch at one point and forgot to take this out

File: larndsim/pixels_from_track_ep.py (hunk 4 above)
Comment (sgasioro) [reply larndsim/pixels_from_track_ep.py:127]: This is mostly in here for comparison with the non-differentiable sim (preserves some of the ordering) -- I think we can probably do this after the fact and switch back to your ep.unique function, but I'll double check

**Review by ynashed — APPROVED** [review]

**Pull request merged**